packages/edit-site/src/components/template-details/index.js

```diff
@@ -31,10 +31,12 @@ export default function TemplateDetails( { template, onClose } ) {
 	);
 	const { revertTemplate } = useDispatch( editSiteStore );
 
+	// TODO: We should update this to filter by template part's areas as well.
 	const browseAllLinkProps = useLink( {
-		// TODO: We should update this to filter by template part's areas as well.
+		canvas: 'view',
 		postType: template.type,
 		postId: undefined,
+		path: '/' + template.type + '/all',
```

Comment:
The `broweAllLink` version from the sidebar navigation component doesn't work in the edit mode.

https://github.com/WordPress/gutenberg/blob/b80d84863e93c2d9b6464718d9e6374f8db0f963/packages/edit-site/src/components/sidebar-navigation-screen-templates/index.js#L63-L65